msg/versioned/BatteryStatus.msg

```diff
@@ -1,74 +1,77 @@
-uint8 cell_count				# Number of cells, 0 if unknown
+uint64 timestamp # Time since system start [us]
+bool connected # Whether or not a battery is connected, based on a voltage threshold
+float32 voltage_v # Battery voltage [V] [@invalid 0]
+float32 current_a # Battery current [A] [@invalid -1]
+float32 current_average_a # Battery current average (for FW average in level flight) [A] [@invalid -1]
+float32 discharged_mah # Discharged amount [mAh] [@invalid -1]
+float32 remaining # Remaining capacity [@range 0,1] [@invalid -1]
+float32 scale # Power scaling factor [@range 1,] [@invalid -1]
+float32 time_remaining_s # Predicted time remaining until battery is empty under previous averaged load [s] [@invalid NaN]
+float32 temperature # Temperature of the battery [°C] [@invalid NaN]
+uint8 cell_count # Number of cells [@invalid 0]
+
 
-uint8 SOURCE_POWER_MODULE = 0
-uint8 SOURCE_EXTERNAL = 1
-uint8 SOURCE_ESCS = 2
-uint8 source					# Battery source
-uint8 priority					# Zero based priority is the connection on the Power Controller V1..Vn AKA BrickN-1
-uint16 capacity					# actual capacity of the battery
-uint16 cycle_count				# number of discharge cycles the battery has experienced
-uint16 average_time_to_empty			# predicted remaining battery capacity based on the average rate of discharge in min
-uint16 serial_number				# serial number of the battery pack
-uint16 manufacture_date				# manufacture date, part of serial number of the battery pack. Formatted as: Day + Month×32 + (Year–1980)×512
-uint16 state_of_health				# state of health. FullChargeCapacity/DesignCapacity, 0-100%.
-uint16 max_error				# max error, expected margin of error in % in the state-of-charge calculation with a range of 1 to 100%
-uint8 id					# ID number of a battery. Should be unique and consistent for the lifetime of a vehicle. 1-indexed.
-uint16 interface_error				# interface error counter
+uint8 source # Battery source [@enum SOURCE]
+uint8 SOURCE_POWER_MODULE = 0 # Power module
+uint8 SOURCE_EXTERNAL = 1 # External
+uint8 SOURCE_ESCS = 2 # ESCs
 
-float32[14] voltage_cell_v			# Battery individual cell voltages, 0 if unknown
-float32 max_cell_voltage_delta			# Max difference between individual cell voltages
+uint8 priority # Zero based priority is the connection on the Power Controller V1..Vn AKA BrickN-1
+uint16 capacity # Actual capacity of the battery [Ah]
+uint16 cycle_count # Number of discharge cycles the battery has experienced
+uint16 average_time_to_empty # Predicted remaining battery capacity based on the average rate of discharge [minutes]
+uint16 serial_number # Serial number of the battery pack
+uint16 manufacture_date # Manufacture date, part of serial number of the battery pack. Formatted as: Day + Month×32 + (Year–1980)×512
+uint16 state_of_health # State of health. FullChargeCapacity/DesignCapacity [%] [@range 0, 100].
+uint16 max_error # Max error, expected margin of error in the state-of-charge calculation [%] [@range 1, 100]
```

Comment:
SHould range start at `0`? i.e. is 0 range allowed?

Reply:
This field seems to be defined in SMBus https://github.com/PX4/PX4-Autopilot/blob/4e17c5496c40adc5c156eb0407b184c74730756a/src/drivers/batt_smbus/batt_smbus.cpp#L157-L158
and it's definition is currently useless in my eyes. Possibly it wants to report some uncertainty/variance on the battery state of charge estimate but that should in my eyes not be a maximum error... seems strange. The variance I'd see as float and based on the same range as the state of charge is reported... but I currently don't have any use for that.

Reply:
Yuck. Note thought that this is same info in new format. So even though this is horrible and needs to be looked at, shouldn't block this PR.